devicetypes/smartthings/fibaro-double-switch.src/fibaro-double-switch.groovy

```diff
@@ -0,0 +1,786 @@
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		sendHubCommand refresh([switchId],includeMeterGet)
+	}
+}
+
+def refresh(endpoints = [1], includeMeterGet = true) {
+	def cmds = []
+	endpoints.each {
+		cmds << [encap(zwave.basicV1.basicGet(), it)]
+		if (includeMeterGet) {
+			cmds << encap(zwave.meterV3.meterGet(scale: 0), it)
+			cmds << encap(zwave.meterV3.meterGet(scale: 2), it)
+		}
+	}
+	delayBetween(cmds, 200)
+}
+
+private resetAll() {
+	childDevices.each { childReset(it.deviceNetworkId) }
+	sendHubCommand reset()
+}
+
+def childReset(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		log.debug "Child reset switchId: ${switchId}"
+		sendHubCommand reset(switchId)
+	}
+}
+
+def reset(endpoint = 1) {
+	log.debug "Resetting endpoint: ${endpoint}"
+	delayBetween([
+			encap(zwave.meterV3.meterReset(), endpoint),
+			encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+			"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		log.debug "encap command: ${cmd} "
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private addChildSwitches(numberOfSwitches) {
+	for (def endpoint : 2..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName + "-${endpoint}"
+			addChildDevice("Child Metering Switch", childDni, device.getHub().getId(), [
+					completedSetup	: true,
+					label			: componentLabel,
+					isComponent		: false
+			])
+		} catch(Exception e) {
+			log.debug "Exception: ${e}"
+		}
+	}
+}
+
+private getParameterMap() {[
+	[
+		name: "Remember device state", key: "rememberDeviceState", type: "boolean",
+		parameterNumber: 1, size: 1, defaultValue: 1,
+		optionInactive: 0, inactiveDescription: "remains switched off after restoring power",
+		optionActive: 1, activeDescription: "restores remembered state after restoring power",
+		description: "This parameter determines how the device will react in the event of power supply failure (e.g. power outage). The parameter is not relevant for outputs set to pulse mode (parameter 150/151 set to 2)."
+	],
+	[
+		name: "LED frame - power limit", key: "ledFrame-PowerLimit", type: "range",
+		parameterNumber: 10, size: 4, defaultValue: 36800,
+		range: "500..36800",
+		 description: "This parameter determines maximum active power. Exceeding it results in the LED frame flashing violet. Function is active only when parameter 11 is set to 8 or 9."
+	],
+	[
+			name: "LED frame - colour when ON", key: "ledFrame-ColourWhenOn", type: "enum",
```

Comment:
The double (and sometimes more) indenting for lists throughout the file gets a bit distracting. It would be much cleaner/more readible as just a single indent. Though this is the only instance of it in `getParameterMap()`.

Reply:
ok, fixed (in a whole file). Please let me know if it looks better now.